### PR TITLE
feat: add auto option for chapters

### PR DESCRIPTION
**Pull request by jakub-roch**
Add auto option for chapters in Video Player (vtt files taken by naming convention, created by chapters editor)

### CODE REVIEW

File: docs/chapters.html [hunk 1]
```diff
@@ -49,7 +49,15 @@
...
           8: 'The attack'
         },
-      })
+      });
+
+      const playerAutoVTT = cloudinary.videoPlayer('player-auto-vtt', {
+        cloud_name: 'demo',
+        publicId: 'snow_horses',
+        chapters: {
+          auto: true,
```
Comment (ehab-cl) [inline docs/chapters.html:58]:
```suggestion
          byName: true,
```

File: docs/chapters.html [hunk 2]
```diff
@@ -49,7 +49,15 @@
           3: 'A dangerous quest',
           8: 'The attack'
         },
-      })
+      });
+
+      const playerAutoVTT = cloudinary.videoPlayer('player-auto-vtt', {
```
Comment (ehab-cl) [inline docs/chapters.html:54]:
```suggestion
      const playerByNameVTT = cloudinary.videoPlayer('player-by-name-vtt', {
```
And where relevant down in the code snippet itself.

File: docs/chapters.html [hunk 3]
```diff
@@ -88,6 +96,19 @@ <h4 class="mb-4">From a config object on <code>source()</code></h4>
       width="600"
     ></video>
 
+    <br><br>
+    <h4 class="mb-4">From an automatically taken VTT file</h4>
+
+    <video
+      id="player-auto-vtt"
```
Comment (ehab-cl) [inline docs/chapters.html:103]:
```suggestion
      id="player-by-name-vtt"
```
And where relevant

File: docs/chapters.html [hunk 4]
```diff
@@ -112,6 +133,15 @@ <h3 class="mt-4">Example Code:</h3>
           class="cld-video-player"
           width="600"
       &gt;&lt;/video&gt;
+
+        &lt;video
+          id="player-auto-vtt"
```
Comment (ehab-cl) [inline docs/chapters.html:138]:
```suggestion
          id="player-by-name-vtt"
```

File: src/plugins/chapters/index.js [hunk 5]
```diff
@@ -85,23 +85,38 @@ const ChaptersPlugin = (function () {
...
 
+  ChaptersPlugin.prototype.getAutoChaptersFileUrl = function getAutoChaptersFileUrl() {
+    const cloudName = this.player.cloudinary.cloudinaryConfig().cloud_name;
+    const currentPublicId = this.player.cloudinary.currentPublicId();
+
+    if (!cloudName || !currentPublicId) {
+      return null;
+    }
+
+    return `https://res.cloudinary.com/${cloudName}/video/upload/v1/${currentPublicId}-chapters.vtt`;
```
Comment (ehab-cl) [inline src/plugins/chapters/index.js:96]: what is the v1 here? i might be wrong but i do not think you need it?

File: src/plugins/chapters/index.js [hunk 6]
```diff
@@ -85,23 +85,38 @@ const ChaptersPlugin = (function () {
...
+  };
+
   /**
    * Bootstrap the plugin.
    */
   ChaptersPlugin.prototype.initializeChapters = function initializeChapters() {
-    if (this.options.url) {
+    const urlByNameFlagProvided = typeof this.options.byName === 'boolean';
+    const chaptersUrlProvided = !!this.options.url;
+    const chaptersUrl = this.options.byName === true ? this.getAutoChaptersFileUrl() : this.options.url;
```
Comment (ehab-cl) [inline src/plugins/chapters/index.js:105]: I would give higher priority to url, this is also easier to read.
```suggestion
    const chaptersUrl = this.options.url ?? this.getAutoChaptersFileUrl();
```

File: src/plugins/chapters/index.js [hunk 7]
```diff
@@ -85,23 +85,38 @@ const ChaptersPlugin = (function () {
     }
   };
 
+  ChaptersPlugin.prototype.getAutoChaptersFileUrl = function getAutoChaptersFileUrl() {
```
Comment (ehab-cl) [inline src/plugins/chapters/index.js:88]:
```suggestion
  ChaptersPlugin.prototype.getChaptersFileUrlByName = function getAutoChaptersFileUrl() {
```

**Review by ehab-cl — APPROVED** [review]
LGTM 💪

File: src/plugins/chapters/index.js (hunk 6 above)
Comment (jakub-roch) [reply src/plugins/chapters/index.js]: no longer the case - this.options can be bool

File: src/plugins/chapters/index.js (hunk 5 above)
Comment (jakub-roch) [reply src/plugins/chapters/index.js]: its necessary to get latest version - docs: https://cloudinary.com/documentation/invalidate_cached_media_assets_on_the_cdn

**Pull request merged**